src/lib.rs

```diff
@@ -255,6 +255,37 @@ mod impl_details {
     }
 }
 
+#[cold]
+fn capacity_overflow() -> ! {
+    panic!("capacity overflow")
+}
+
+trait OptExt<T> {
+    fn unwrap_cap_overflow(self) -> T;
+}
+
+impl<T> OptExt<T> for Option<T> {
+    fn unwrap_cap_overflow(self) -> T {
+        match self {
+            Some(val) => val,
+            None => capacity_overflow(),
+        }
+    }
+}
+
+trait ResExt<T> {
```

Comment:
**question:** Is the separate traits for Opt and Res actually needed? Couldn't this just be a singular `UnwrapCapacityOverflow` trait that is implemented for both `Option<T>` and for `Result<T, E>`?

Reply:
They weren't needed, I guess they were a leftover from another design I had to fix this, fixed anyway.